### DISCUSSION TITLE
My Subscription "Isn't Found"

### DISCUSSION THREAD

**Discussion by Schidhuv** [homebridge-simplisafe3/homebridge-simplisafe3#201] [Q&A] [answered]
I am completely new to HOOBS and don't know much about computer programing and what not, so any assistance would be greatly appreciated!

I am trying to connect SimpliSafe with my Apple Homekit.  When I input my username and password for the simplisafe3 plugin configuration, I get the following error in the log:
__________
3/1/2021, 2:23:30 PM [Home Alarm] Initial accessories refresh failed with error: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/simplisafe.js:517:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async SS3Platform.refreshAccessories (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/index.js:157:26)
__________

I know this is the right username and password because it works on all my other devices... and SimpliSafe also sent me a verification email when I first connected HOOBS/this plug-in with said username and password (and of course I clicked "allow the the connection").  But I still get the error above.  Am I missing anything?  Thanks in advance!

Chris

**Comment by shamoon** [comment]
Are you paying for the interactive monitoring plan? It’s required to use their apps etc., and thus this plug-in

**Reply by Schidhuv** [reply]
I should have included that above.  Yes, we are paying for the plan.

**Reply by shamoon** [reply]
Hmm, your config (no PW/login) and logs would be helpful, you can enable debug on this plug-in but I don’t use nor officially support HOOBS (since they’re kinda just profiting off other peoples open source work) so I’m not sure how to do enable it for HOOBS itself but that’s needed for full logs. For homebridge you run it with the `-D` option (or enable that via the UI), I’m guessing it’s same/similar. See https://github.com/homebridge-simplisafe3/homebridge-simplisafe3#debug

**Reply by Schidhuv** [reply]
I'm sorry, I'm not understanding everything you are saying (like I said, I'm very new to all this!).  Are you saying you would be able to answer my question if I could provide the whole log?    Is there a better solution besides HOOBS that a super novice could use?

**Reply by shamoon** [reply]
Yes, that info might help.

HOOBS is derived from an open source project called Homebridge. Since the two separated Homebridge has become much easier to setup but still may not be quite as easy as HOOBS. But HOOBS still uses 100% plugins created by the community, for free. The other advantage of HOOBS is (I think) you are entitled to some support, unless they just punt you over here, but perhaps you can reach out to them.

If you’re curious to try Homebridge itself you can check out https://github.com/homebridge/homebridge/wiki .

Ps. Have you restarted it?

**Reply by Schidhuv** [reply]
I have restarted ( a few times.  I will put the log below from the latest restart:

3/1/2021, 7:13:18 PM HOOBS listening on port 80.
3/1/2021, 7:13:25 PM Loaded plugin 'homebridge-simplisafe3'
3/1/2021, 7:13:25 PM [3/1/2021, 5:13:25 PM] Registering platform 'homebridge-simplisafe3.SimpliSafe 3'
3/1/2021, 7:13:25 PM Loading 1 platforms...
3/1/2021, 7:13:25 PM [Home Alarm] Initializing homebridge-simplisafe3.SimpliSafe 3 platform...
3/1/2021, 7:13:26 PM Bridge is running on port 51826.
3/1/2021, 7:13:26 PM [Home Alarm] An error occurred while refreshing accessories: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/simplisafe.js:517:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async SS3Platform.refreshAccessories (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/index.js:157:26)
3/1/2021, 7:13:26 PM [Home Alarm] SimpliSafe login failed with error: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/simplisafe.js:517:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async SS3Platform.refreshAccessories (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/index.js:157:26)
3/1/2021, 7:13:27 PM [Home Alarm] An error occurred while refreshing accessories: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/simplisafe.js:517:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async SS3Platform.refreshAccessories (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/index.js:157:26)
3/1/2021, 7:13:27 PM [Home Alarm] Initial accessories refresh failed with error: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/simplisafe.js:517:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async SS3Platform.refreshAccessories (/home/hoobs/.hoobs/node_modules/homebridge-simplisafe3/index.js:157:26)

**Reply by shamoon** [reply]
I am wondering if there is something specific to your account, its hard for me to figure out without looking at it directly though, and I am wondering if you can make a tiny edit to the code and then send me the output? What I'd like to see (with private info redacted) would be to add `this.log.debug(data)` to the function `getSubscriptions()` in simplisafe.js. This would really just be editing a text file and adding that text at line number 492, then restarting but if that makes no sense to you perhaps you can reach out to HOOBS and they can help you do that and get back to me? Otherwise I'd have to release a whole new version just to get that info.

Obviously I cant test your account directly myself.

**Comment by badxhabit** [comment]
Hi,

I would like to bump this thread because I have had the exact same issue with my system for almost a year now and I finally had some spare time to debug this issue and found that my "accountNumber" was never being set in the `getSubscriptions()` function on Line Number: 307. So, I commented out the following code on line number: 297, and everything works now!!

`//subscriptions = subscriptions.filter(s => s.location.account === this.accountNumber);`

What does this line of code do and why is it there? It appears to just be breaking the whole setup. Can we please remove it? I can send a PR, if needed.

Thanks,
Sonny

**Reply by shamoon** [reply]
Hi Sonny, that doesnt exactly make sense, the line you are referring to is wrapped in an [`if (this.accountNumber) {`](https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/blob/49c023dfd71287d3d2788edd4bcd3f20bb52d9bf/src/simplisafe.js#L251), in other words that line only runs if an `accountNumber` is set, which only happens if you are specifiying a subscriptionId in the config, see https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/blob/49c023dfd71287d3d2788edd4bcd3f20bb52d9bf/src/index.js#L48

The purpose of that line is to allow people to filter for a specific subscription which was requested by another user at some point. So yea, if you arent specifying a `subscriptionId` in your config I dont think that would be the issue...

**Comment by badxhabit** [comment]
Hi Michael,

Ya, I found it strange because I have a subscriptionId, which is in my config.json file (also was set by the Authentication piece in Config-UI-X, thank you for that!)

I don't understand why it would be blank for me. I did have a different subscription at some point, I believe it was because of the free trial, and then my new/current subscription is the "2nd" one.

Either way, I don't think this is filtering correctly because it wasn't finding a match and commenting out that line fixed it.

- Sonny

**Comment by badxhabit** [comment]
However, I just remembered that I didn't recall seeing "accountNumber" being set anywhere. 

I recall it being set as an else:
`if (subs.length == 1) {
          subscriptionId = subs[0].sid;
        } else if (subs.length == 0) {
          throw new Error('No matching subscriptions found. Check your account and ensure you have an active subscription.');
        } else {
          let accountNumbers = subs.map(s => s.location.account);
          throw new Error(`Multiple subscriptions found. Edit your config.json file and add a parameter called "subscriptionId": "YOUR AC$
        }`

But my homebridge would always error out on me on the condition above ` } else if (subs.length == 0) {
          throw new Error('No matching subscriptions found. Check your account and ensure you have an active subscription.');`

This still doesn't make sense because when I `this.log(this.accountNumber);` on line 295 (above the if condition), I get my subscriptionId that was put in my config.json. Could SimpliSafe have changed the structure of the response back? **(s.location.account)**

**Reply by shamoon** [reply]
If you add a line to log the response of the api call in `getSubscriptions` what do you get? Eg after `let data = await this.request`… what does `data` return for you?

**Reply by shamoon** [reply]
Actually would be good to see one level deeper too, e.g.
```
this.log.debug(data);
this.log.debug(data.subscriptions[0].location);
```

For me they're as expected. 

Btw, are you using the correct account number? You can find it on https://webapp.simplisafe.com/#/account under *Protected Locations*

**Reply by badxhabit** [reply]
Interesting. My account number is correct.

Here are the following outputs from the commands (redacted any sensitive information):

```
this.log.debug(data);

[Home Alarm] {
  subscriptions: [
    {
      uid: xxxxxxxx,
      sid: xxxxxxxx,
      sStatus: 7,
      activated: 1620419931,
      planSku: 'SSSLFCAM',
      planName: 'Self Monitoring with Camera Recording',
      price: 9.99,
      currency: 'USD',
      country: 'US',
      expires: 1640370870,
      canceled: 0,
      extraTime: 0,
      creditCard: [Object],
      time: 2628000,
      paymentProfileId: 'xxxxxx',
      upgradeStatus: 0,
      features: [Object],
      status: [Object],
      subscriptionFeatures: [Object],
      dispatcher: 'cops',
      dcid: 7,
      activationCode: '',
      systemVersion: 20,
      orderProductId: 0,
      location: [Object],
      pinUnlocked: true,
      billDate: 1640370870,
      billInterval: 2628000,
      pinUnlockedBy: 'pin',
      autoActivation: null
    },
    {
      uid: xxxxxxx,
      sid: xxxxxxx,
      sStatus: 0,
      activated: 0,
      planSku: 'SSEDSM2',
      planName: 'Interactive Monitoring',
      price: 24.99,
      currency: 'USD',
      country: 'US',
      expires: 0,
      canceled: 0,
      extraTime: 2628000,
      creditCard: [Object],
      time: 2628000,
      paymentProfileId: 'xxxxxxx',
      features: [Object],
      status: [Object],
      subscriptionFeatures: [Object],
      dispatcher: 'cops',
      dcid: 7,
      activationCode: 'xxxxxx',
      systemVersion: 20,
      orderProductId: xxxxxx,
      location: [Object],
      pinUnlocked: false,
      billDate: 2628000,
      billInterval: 2628000,
      pinUnlockedBy: '',
      autoActivation: null
    }
  ]
}

this.log.debug(data.subscriptions[0].location);

[12/10/2021, 9:43:50 AM] [Home Alarm] {
  sid: xxxxxx,
  uid: xxxxxx,
  lStatus: 0,
  account: 'xxxxxxxx',
  street1: 'xxxxxxx',
  street2: '',
  locationName: '',
  city: 'xxxxxxx',
  county: 'xxxxxxx',
  state: 'xx',
  zip: 'xxxxx',
  country: 'US',
  crossStreet: '',
  notes: '',
  residenceType: x,
  numAdults: x,
  numChildren: x,
  locationOffset: -480,
  safeWord: 'xxxxxxxx',
  signature: 'xxxxx xxxxxxx',
  timeZone: 3,
  primaryContacts: [
    {
      name: 'xxxx xxxx',
      firstName: 'xxxxxx',
      lastName: 'xxxxxx',
      phone: 'xxxxxxx'
    },
    {
      name: 'xxxxxxx',
      firstName: 'xxxxx',
      lastName: 'xxxxx',
      phone: 'xxxxxx'
    }
  ],
  secondaryContacts: [
    { name: 'xxxxxx', phone: 'xxxxxx' },
    { name: '', phone: '' },
    { name: '', phone: '' },
    { name: '', phone: '' },
    { name: '', phone: '' }
  ],
  copsOptIn: true,
  certificateUri: 'https://simplisafe.com/account2/xxxxxxx/alarm-certificate/xxxxxxx',
  nestStructureId: '',
  system: {
    serial: 'xxxxxxx',
    alarmState: 'HOME',
    alarmStateTimestamp: 1639119738,
    isAlarming: false,
    version: 3,
    capabilities: {
      setWifiOverCell: true,
      setDoorbellChimeVolume: true,
      outdoorBattCamera: true
    },
    temperature: null,
    exitDelayRemaining: 60,
    cameras: [ [Object] ],
    connType: 'wifi',
    stateUpdated: 1639157029,
    messages: [],
    powerOutage: false,
    lastPowerOutage: 1637968811,
    lastSuccessfulWifiSSID: 'xxxxxx',
    lastSuccessfulWifiTS: 1639157029,
    isOffline: false
  }
}
```

**Reply by badxhabit** [reply]
I went ahead and added the code below after ` let subscriptions = data.subscriptions.filter(s => [7, 10, 20].includes(s.sStatus));`:

This is what I got:

` this.log.debug(s.location.account);
`[12/10/2021, 9:59:41 AM] [Home Alarm] An error occurred while refreshing accessories: ReferenceError: s is not defined
    at SimpliSafe3.getSubscriptions (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:295:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SimpliSafe3.getSubscription (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:317:20)
    at SS3Platform.refreshAccessories (/usr/lib/node_modules/homebridge-simplisafe3/index.js:168:26)`

`this.log.debug(this.accountNumber);`
`[12/10/2021, 9:59:41 AM] [Home Alarm] Initial accessories refresh failed with error: ReferenceError: s is not defined
    at SimpliSafe3.getSubscriptions (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:295:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SimpliSafe3.getSubscription (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:317:20)
    at SS3Platform.refreshAccessories (/usr/lib/node_modules/homebridge-simplisafe3/index.js:168:26)`

It appears that `s` isn't defined in the above line, but my sStatus is 7.

**Reply by shamoon** [reply]
Sorry no, `s` is a local variable to that function, it wouldn't be available after that line.

Your `subscription` object looks just fine, it has the correct `sStatus` and has the expected `location` property which has the expected `account` property.

So the account number you have in config is the same what you got from `location.account`? (For me its the serial on my base station, i.e. it also shows under location > system > serial

**Reply by badxhabit** [reply]
The config was populated from the auto-authentication process in Config-UI-X.

So there is no "account number", just subscriptionId in my config.js.

`{
            "name": "Home Alarm",
            "cameras": true,
            "subscriptionId": "xxxxxxx",
            "sensorRefresh": 15,
            "persistAccessories": true,
            "resetSimpliSafeId": false,
            "cameraOptions": {
                "enableHwaccelRpi": true
            },
            "platform": "homebridge-simplisafe3.SimpliSafe 3"
},`

**Reply by shamoon** [reply]
Authentication doesnt fill out that `subscriptionId` field automatically, thats something you would have done yourself at some point. Yes, `subscriptionId` is the same as `accountNumber`, I just meant to make sure its the correct one in your config (same as the one that you got towards the bottom in your [comment](https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/201#discussioncomment-1787237) above), sounds like it is. Basically if its set to the right account number, I see no reason why that code would fail.

In other words if you change it to
```
    let subscriptions = data.subscriptions.filter(s => [7, 10, 20].includes(s.sStatus));
    this.log.debug(`Detected account number ${this.accountNumber}`);
    this.log.debug('subscriptions before filter:');
    this.log.debug(subscriptions);

    if (this.accountNumber) {
      subscriptions = subscriptions.filter(s => {
        this.log.debug(`Checking subscription ${s.planName} with sid ${s.sid}`);

        if (s.location && s.location.account) {
          this.log.debug(`Subscription sid ${s.sid} has account number ${s.location.account}`);
          if (s.location.account === this.accountNumber) this.log.debug(`Subscription found! Subscription sid ${s.sid} has account number ${s.location.account} which matches config account number ${this.accountNumber}`);
        }

        return s.location.account === this.accountNumber;
      });
    }

    this.log.debug('subscriptions after filter:');
    this.log.debug(subscriptions);
```

it should illustrate this.

**Reply by badxhabit** [reply]
Hey Michael,

After running the script, I see that my subscriptionId and Account Number are different:

[12/13/2021, 10:25:31 AM] [Home Alarm] Checking subscription Self Monitoring with Camera Recording with sid xxxxxx6
[12/13/2021, 10:25:31 AM] [Home Alarm] Subscription sid xxxxxx6 has account number xxxxxx2
[12/13/2021, 10:25:31 AM] [Home Alarm] subscriptions after filter:
[12/13/2021, 10:25:31 AM] [Home Alarm] []
[12/13/2021, 10:25:31 AM] [Home Alarm] An error occurred while refreshing accessories: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:334:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SS3Platform.refreshAccessories (/usr/lib/node_modules/homebridge-simplisafe3/index.js:168:26)
[12/13/2021, 10:25:31 AM] [Home Alarm] SimpliSafe login failed with error: Error: No matching subscriptions found. Check your account and ensure you have an active subscription.
    at SimpliSafe3.getSubscription (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:334:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SS3Platform.refreshAccessories (/usr/lib/node_modules/homebridge-simplisafe3/index.js:168:26)
[12/13/2021, 10:25:31 AM] [Home Alarm] See the plugin README for more information on authenticating with SimpliSafe.
[12/13/2021, 10:25:34 AM] [Home Alarm] Detected account number xxxxxx6
[12/13/2021, 10:25:34 AM] [Home Alarm] subscriptions before filter:
[12/13/2021, 10:25:35 AM] [Home Alarm] [
  {

**Reply by shamoon** [reply]
`sid` is not actually subscriptionId, I know it’s confusing. Your config should have the number above that ends in 2 (account number). If that’s the number in your config the code should work. Is it?

**Reply by shamoon** [reply]
Oh actually I see your config has a different one. I’m guessing if you change it that fixes it? So it seems like on your account your account number is **not** the serial number on the base station? I assume it’s different cause of the trial. I can update the documentation

**Reply by badxhabit** [reply]
Ah, I see.

I guess the verbiage of subscriptionId vs. accountNumber is a bit confusing. When I first installed the plug-in, I was told to get the "subscriptionId", which I assumed was the ID of the subscription I am paying for, not the Account Number found on the actual unit, even though its pointed out in the caption.

Is there a reason why the field is "Subscription Id", and not "Account Number"?

**Reply by shamoon** [reply]
I didnt initially name that variable. I can see how this could be confusing. I will update the settings UI and documentation. 

Glad things are sorted...

**Reply by badxhabit** [reply]
Thank you so much for your patience and understanding. I LOVE THE PLUG-IN!! Makes the systems soooooooo much more useful with the automations!

**Comment by shamoon** [comment, marked as answer]
For anyone else finding this, the best way to get the correct account number is under the [SimpliSafe web control panel > View Account](https://webapp.simplisafe.com/#/account) and look for **account #** for the correct plan.
